xray/utils/resultwriter.go

```diff
@@ -203,55 +214,70 @@ func convertToVulnerabilityOrViolationSarif(run *sarif.Run, jsonTable *formats.S
 	return convertVulnerabilitiesToSarif(jsonTable, run, markdownOutput)
 }
 
-func convertToIacOrSecretsSarif(run *sarif.Run, jsonTable *formats.SimpleJsonResults, markdownOutput bool) error {
-	var err error
+func convertToSourceCodeResultSarif(run *sarif.Run, jsonTable *formats.SimpleJsonResults, markdownOutput bool) (err error) {
 	for _, secret := range jsonTable.Secrets {
-		properties := getIacOrSecretsProperties(secret, markdownOutput, true)
+		properties := getSourceCodeProperties(secret, markdownOutput, Secrets)
 		if err = addPropertiesToSarifRun(run, &properties); err != nil {
-			return err
+			return
 		}
 	}
 
 	for _, iac := range jsonTable.Iacs {
-		properties := getIacOrSecretsProperties(iac, markdownOutput, false)
+		properties := getSourceCodeProperties(iac, markdownOutput, IaC)
 		if err = addPropertiesToSarifRun(run, &properties); err != nil {
-			return err
+			return
 		}
 	}
-	return err
+
+	for _, sast := range jsonTable.Sast {
+		properties := getSourceCodeProperties(sast, markdownOutput, Sast)
+		if err = addPropertiesToSarifRun(run, &properties); err != nil {
+			return
+		}
+	}
+	return
 }
 
-func getIacOrSecretsProperties(secretOrIac formats.IacSecretsRow, markdownOutput, isSecret bool) sarifProperties {
-	file := strings.TrimPrefix(secretOrIac.File, string(os.PathSeparator))
+func getSourceCodeProperties(sourceCodeIssue formats.SourceCodeRow, markdownOutput bool, scanType ScanType) sarifProperties {
+	file := strings.TrimPrefix(sourceCodeIssue.File, string(os.PathSeparator))
 	mapSeverityToScore := map[string]string{
```

Comment:
can we just add this scores to the sevirities map?

Reply:
@sverdlov93 I don't know what those scores represent, do you know?

Reply:
for the security tab in GitHub you need to specify a score of 0-10 it will then map the score to the severity level.